R/estimateDiversity.R

```diff
@@ -184,7 +194,123 @@ setMethod("estimateDiversity", signature = c(x = "SummarizedExperiment"),
     }
 )
 
+#' @rdname estimateDiversity
+#' @export
+setMethod("estimateDiversity", signature = c(x="TreeSummarizedExperiment"),
+    function(x, abund_values = "counts",
+             index = c("shannon","gini_simpson","inverse_simpson",
+                       "coverage", "fisher", "faith"),
+             name = index, ..., BPPARAM = SerialParam()){
+        
+        # Gets the tree 
+        tree <- rowTree(x)
```

Comment:
I was thinking if this should be removed, because estimateFaith is already doing this same job

Reply:
But that you can use `tree` in the check in line 210 anymore...

Reply:
That's true, I take one step back

Reply:
reverted to previous state